projects/dotcms-field-elements/src/components/dot-date/dot-date.e2e.ts

```diff
@@ -1,131 +1,385 @@
 import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
 import { EventSpy } from '@stencil/core/dist/declarations';
+import { dotTestUtil } from '../../utils';
 
 describe('dot-date', () => {
     let page: E2EPage;
     let element: E2EElement;
-    let input: E2EElement;
-    let spyStatusChangeEvent: EventSpy;
-    let spyValueChange: EventSpy;
+    let inputCalendar: E2EElement;
 
     beforeEach(async () => {
         page = await newE2EPage({
-            html: `
-              <dot-date
-                    label="Date:"
-                    name="date01"
-                    value="2019-01-20"
-                    hint="date hint"
-                    required
-                    required-message="Required Date"
-                    validation-message="Invalid Date Range"
-                    min="2019-01-01"
-                    max="2019-10-30"
-                    step="2"
-                ></dot-date>`
-        });
-
-        spyStatusChangeEvent = await page.spyOnEvent('statusChange');
-        spyValueChange = await page.spyOnEvent('valueChange');
+            html: `<dot-date></dot-date>`
+        });
         element = await page.find('dot-date');
-        input = await page.find('input');
+        inputCalendar = await page.find('dot-input-calendar ');
     });
 
-    it('should render', () => {
-        // tslint:disable-next-line:max-line-length
-        const tagsRenderExpected = `<dot-date label=\"Date:\" name=\"date01\" value=\"2019-01-20\" hint=\"date hint\" required=\"\" required-message=\"Required Date\" validation-message=\"Invalid Date Range\" min=\"2019-01-01\" max=\"2019-10-30\" step=\"2\" class=\"dot-valid dot-pristine dot-untouched dot-required hydrated\"><div class=\"dot-field__label\"><label for=\"dot-date01\">Date:</label><span class=\"dot-field__required-mark\">*</span></div><dot-input-calendar type=\"date\" required-message=\"Required Date\" validation-message=\"Invalid Date Range\" class=\"hydrated\"><input id=\"dot-date01\" required=\"\" type=\"date\" min=\"2019-01-01\" max=\"2019-10-30\" step=\"2\"></dot-input-calendar><span class=\"dot-field__hint\">date hint</span></dot-date>`;
-        expect(element.outerHTML).toBe(tagsRenderExpected);
-    });
+    describe('render CSS classes', () => {
+        it('should be valid, untouched & pristine on load', () => {
+            expect(element).toHaveClasses(dotTestUtil.class.empty);
+        });
 
-    it('should be valid, touched & dirty on value change', async () => {
-        await input.press('2');
-        await page.waitForChanges();
-        expect(element.classList.contains('dot-valid')).toBe(true);
-        expect(element.classList.contains('dot-dirty')).toBe(true);
-        expect(element.classList.contains('dot-touched')).toBe(true);
-    });
+        it('should be valid, touched & dirty when filled', async () => {
+            inputCalendar.triggerEvent('_statusChange', {
```

Comment:
Why don't you create a function where you just pass something like `triggerStatusChange(false, true, true)` because there is a lot of repeating code here.